### ISSUE TITLE
Train on Non-Square Images

### ISSUE THREAD

**Issue by anujdutt9** [ultralytics/yolov5#4116]
## ❔Question
Hi. I am trying to train the YOLOv5s on Crowd Human dataset, but instead of using the default 640x640 square image, I would like to use 512x320 image, width and height are a multiple of 32. What's the best approach of doing this?

## Additional context
I have seen some issues answered here saying to use a `--rect` flag, but how do I pass in my image size as the `train.py` accepts only a single size for `[train, test]`?
Also, can I pass in these images as is, without using the `--rect` flag as I would like to use the mosaic augmentation?

Thanks

**Comment by glenn-jocher** [comment]
@anujdutt9 
```
python train.py --img 512 --rect
```

**Comment by anujdutt9** [comment]
Hi @glenn-jocher Thanks for your reply. I did try this and but it looks like the image has my provided width but not the height, as there is no way to provide it here. I did check out your discussion [here](https://github.com/ultralytics/yolov3/issues/232) and it looks like you take one dimension as constant and the other dimension is taken as closest as possible to nearest multiple of 32. But what if I need to keep both the image height and width as constant? Like 512x320. Both are multiples of 32 and hence the resulting model should be able to take this as input. But instead the resulting model, even after using `--rect` flag takes the input as 512x512.
Should I change the code to take both height and width as input instead of just one dimension?
Thanks
